setup.py

```diff
@@ -1,34 +1,70 @@
+import json
 import os
 import platform
+from pathlib import Path
 from setuptools import setup
 from pybind11.setup_helpers import Pybind11Extension, build_ext
 
-extra_compile_args = [
-    "-MMD",
-    "-MP",
-    "-Wextra",
-    "-Wpedantic",
-    "-Wall",
-    "-O3",
-    "-DASIO_STANDALONE",
-    "-DNDEBUG",
-]
 extra_link_args = [
     "-lpthread",
     "-lssl",
     "-lcrypto",
 ]
+extra_compile_args = [
+        "-MMD",
+        "-MP",
+        "-Wextra",
+        "-Wpedantic",
+        "-Wall",
+        "-O3",
+        "-DASIO_STANDALONE",
+        "-DNDEBUG",
+    ]
 include_dirs = [os.path.join("vroom", "src")]
+libraries = []
+library_dirs = []
+
+# try conan dependency resolution
+conanfile = tuple(Path().rglob('conanbuildinfo.json'))
+if conanfile:
+    print("INFO: Using conan to resolve dependencies.")
+    with open(conanfile[0]) as f:
+        conan_deps = json.load(f)['dependencies']
+    for dep in conan_deps:
+        include = dep['include_paths']
+        libs = dep['libs']
+        system_libs = dep['system_libs']
+        libdirs = dep['lib_paths']
+
+        include_dirs.extend(include)
+        if libs:
+            libraries.extend(libs)
+        if system_libs:
+            libraries.extend(system_libs)
+        if libdirs:
+            library_dirs.extend(libdirs)
+else:
+    print('WARN: Conan not installed and/or no conan build detected. Assuming dependencies are installed.')
 
 if platform.system() == "Darwin":
     # Homebrew puts include folders in weird places.
     include_dirs.append("/usr/local/opt/openssl@1.1/include")
     extra_link_args.insert(0, "-L/usr/local/opt/openssl@1.1/lib")
```

Comment:
actually, using conan on mac for ci would remove these lines as well. which are not very robust I think. brew is pretty updated always I found, some users might actually have 3.x soon? anyways, more of a works-now-why-change thing. but maybe good to keep in mind in worst case